src/fs/dvfs/dvfs.c

```diff
@@ -225,6 +231,10 @@ int dvfs_open(const char *path, struct file *desc, int mode) {
 		.f_dentry = lookup.item,
 		.f_inode  = i_no,
 		.f_ops    = lookup.item->d_sb->sb_fops,
+		.f_idesc  = {
+			.idesc_amode = FS_MAY_READ | FS_MAY_WRITE,
```

Comment:
Yes, mode handling is awful. It requiers new branch I think. Will do it after merge.